tests/components/nest/sensor_sdm_test.py

```diff
@@ -0,0 +1,248 @@
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {},
+            },
+            auth=None,
+        )
+    }
+    await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is None
+
+    humidity = hass.states.get("sensor.my_sensor_humidity")
+    assert humidity is None
+
+
+async def test_device_name_from_structure(hass):
+    """Test a device without a custom name, inferring name from structure."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 25.2,
+                    },
+                },
+                "parentRelations": [
+                    {"parent": "some-structure-id", "displayName": "Some Room"}
+                ],
+            },
+            auth=None,
+        )
+    }
+    await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.some_room_temperature")
+    assert temperature is not None
+    assert temperature.state == "25.2"
+
+
+async def test_event_updates_sensor(hass):
+    """Test a pubsub message received by subscriber to update temperature."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {
+                    "sdm.devices.traits.Info": {
+                        "customName": "My Sensor",
+                    },
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 25.1,
+                    },
+                },
+            },
+            auth=None,
+        )
+    }
+    subscriber = await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is not None
+    assert temperature.state == "25.1"
+
+    # Simulate a pubsub message received by the subscriber with a trait update
+    event = EventMessage(
+        {
+            "eventId": "some-event-id",
+            "timestamp": "2019-01-01T00:00:01Z",
+            "resourceUpdate": {
+                "name": "some-device-id",
+                "traits": {
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 26.2,
+                    },
+                },
+            },
+        },
+        auth=None,
+    )
+    subscriber.receive_event(event)
+    await hass.async_block_till_done()  # Process dispatch/update signal
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is not None
+    assert temperature.state == "26.2"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)